app/models/version.rb

```diff
@@ -1,14 +1,28 @@
 # frozen_string_literal: true
 
-# Represents a version of a digital object.
+# Represents a version of a digital object with associated metadata.
 # All workflow steps occur on to a particular version.
 class Version
-  def initialize(druid:, version:)
+  def initialize(druid:, version:, metadata: nil)
     @druid = druid
     @version_id = version
+    @metadata = metadata # this is a hash of metadata values to be stored in the VersionMetadata table
   end
 
-  attr_reader :druid, :version_id
+  attr_reader :druid, :version_id, :metadata
+
+  def update_metadata
+    # if no metadata passed in (nil), do nothing
+    return unless metadata
+
+    # if json metadata is passed in but is empty, delete the version metadata record
+    if metadata == '{}'
```

Comment:
Line 9 suggests `metadata` is a hash but line 19 checks if `metadata` is a (particular) string. I'm guessing one or the other could use a tweak.

Reply:
Yes, good catch.  Originally it was a hash, but was later tweaked to be a JSON string to better match what is stored in the database.  This makes for less back and forth conversions from hash to json.  I tweaked the comments to make it clearer.